src/rtpi_internal.h

```diff
@@ -6,22 +6,43 @@
 
 #include <linux/futex.h>
 
-#include "rtpi.h"
-
-typedef struct pi_mutex {
-	__u32	futex;
-	__u32	flags;
-} pi_mutex_t;
-
-typedef struct pi_cond {
-	__u32		cond;
-	__u32		flags;
-
-	pi_mutex_t	priv_mut;
-	__u32		wake_id;
-	__u32		pending_wake;
-	__u32		pending_wait;
-	pi_mutex_t	*mutex;
-} pi_cond_t;
+/*
+ * PI Mutex
+ */
+union pi_mutex {
+	struct {
+		__u32	futex;
+		__u32	flags;
+	};
+	__u32 pad[4];
+};
+
+#define PI_MUTEX_INIT(f) { .futex = 0, .flags = f }
+
+/*
+ * PI Cond
+ */
+union pi_cond {
+	struct {
+		__u32		cond;
+		__u32		flags;
+
+		union pi_mutex	priv_mut;
+		__u32		wake_id;
+		__u32		pending_wake;
+		__u32		pending_wait;
+		union pi_mutex	*mutex;
+	};
+	__u32 pad[24];
```

Comment:
Similarly here, why 24?

Reply:
Same as pi_mutex the initial intent was to make it double the current structure size for lack of a better number. Although recomputing the sizes looks like I've padded it up from 20 to 24. I can make this more precise if we decide to continue with this approach.

Similar to pi_mutex this structure might be better if it is cache line aligned/sized. If we go with one cache line per pi_mutex then the pi_cond will have to span two cache lines (due to the embedded private mutex).

Also a bit of a preview, in future patches this structure will go down in size i.e. I think we can do away with the internal mutex because we are already mandating holding the user mutex before calling wait/signal.